packages/core/test/SignUp.ts

```diff
@@ -80,5 +84,38 @@ describe('Signup', function () {
                 'ReachedMaximumNumberUserSignedUp'
             )
         })
+
+        it('setUsername should succeed', async () => {
+            const tx = await unirepSocialContract.setUsername(
+                config.NUM_EPOCH_KEY_NONCE_PER_EPOCH,
+                oldUsername,
+                newUsername,
+                { value: DEFAULT_ATTESTING_FEE }
+            )
+            const receipt = await tx.wait()
+
+            expect(receipt.status).equal(1)
+            const isClaimed = await unirepSocialContract.usernames(newUsername)
+
+            expect(isClaimed, 'This username has not been updated').to.be.true
+        })
+
+        it('should fail if a username is double registered', async () => {
+            const accounts = await ethers.getSigners()
+            await unirepSocialContract
```

Comment:
This test should have a failure condition. It should be completely separate from the last test (don't use the same usernames).

The flow should be:
```
Register username u1 to epoch key e1
Try to register username u1 to epoch key e2 (expecting a failure)
```